pmd-java/src/main/java/net/sourceforge/pmd/lang/java/symbols/internal/asm/AsmSymbolResolver.java

```diff
@@ -124,6 +125,7 @@ void registerKnown(@NonNull String internalName, ClassStub innerClass) {
         knownStubs.put(internalName, softRef);
     }
 
+    @SuppressWarnings("PMD.CompareObjectsWithEquals") // SoftClassReference
```

Comment:
Technically there would be no difference between `prev != failed` or `!Objects.equals(prev, failed)`, since SoftClassReference doesn't override `equals()`.